### ISSUE TITLE
Consistency between Cache and SQL Database

### ISSUE THREAD

**Issue by luantranminh** [luantranminh/til#149]
https://yunpengn.github.io/blog/2019/05/04/consistent-redis-sql/

**TL;DR:** With cache, performance is improved, this approach brings a huge headache that we do not have a single source of truth anymore. Now, the same peace of data would be stored in two places. How can we ensure the consistency between the data stored in cache and the data stored in SQL database?

Remarkable solutions:
- Write through: it requires Cache built-in functions to perform operations (create, update, delete) **synchronically**.
- Write behind: uses the **message queue** to sync with the database **asynchronously**.

Alibaba approach: https://github.com/alibaba/canal
![image](https://user-images.githubusercontent.com/25551568/213369047-746e71ef-5308-4ea7-9816-6972e454df0f.png)
Treat others data layers (Kafka, Redis ...) as a slave. Uses `canal` to read from `binlog` and sync with other slaves.
